File: source/server/Karamem0.Preddy.Web/Services/Entities/TweetSummary.cs
//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/preddy/blob/main/LICENSE
//

using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Karamem0.Preddy.Services.Entities
{

    public class TweetSummary
    {

        public TweetSummary()
        {
        }

        public DateTime? MinDate { get; set; }

        public DateTime? MaxDate { get; set; }

        public IReadOnlyList<TweetSummaryItem>? Items { get; set; }

    }

}

File: source/client/src/types/tweet-summary.ts
//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/preddy/blob/main/LICENSE
//

export interface TweetSummary {
  minDate: Date,
  maxDate: Date,
  items: TweetSummaryItem[]
}

export interface TweetSummaryItem {
  date: Date,
  forecast: number,
  actual: number
}

export interface TweetSummaryDto {
  minDate: string,
  maxDate: string,
  items: TweetSummaryItemDto[]
}

export interface TweetSummaryItemDto {
  date: string,
  forecast: number,
  actual: number
}
